src/scope/server/tracks.py

```diff
@@ -104,10 +144,13 @@ async def recv(self) -> VideoFrame:
         self.initialize_output_processing()
         while self.input_task_running:
             try:
-                # Update FPS from FrameProcessor
-                if self.frame_processor:
+                # Update FPS: prefer input FPS measurement, fall back to FrameProcessor
+                input_fps = self._get_input_fps()
```

Comment:
I don't think we want to always prefer input FPS because the output FPS would be too high for the pipeline when input FPS > pipeline FPS. We don't actually handle that scenario right now (typically either try to throttle the sender so they reduce their input FPS if possible or drop input frames if former is not possible), but generally I think the safe conditional here would be to only use the input FPS as the output FPS if it is lower than the pipeline FPS. See my other comment for a short code example.